openraft/src/core/admin.rs

```diff
@@ -217,7 +220,7 @@ impl<'a, D: AppData, R: AppDataResponse, N: RaftNetwork<D>, S: RaftStorage<D, R>
 
         // Caveat: membership must be updated before commit check is done with the new config.
         self.core.effective_membership = EffectiveMembership {
-            log_id: self.core.last_log_id,
+            log_id: self.core.last_log_id.expect("raft core last_log_id is uninitialized"),
```

Comment:
Here it's better to use the `log_id` of the returned entry in `res`.

There is another issue that updating the `effective_membership` at line 222 should be placed after checking `res`, e.g. at line 242